CMSgov/easi-app
# Rework draft business case submission to CEDAR Intake

# EASI-1538

## Changes and Description

- Re-applies changes from https://github.com/CMSgov/easi-app/pull/1785 to submit draft business case data to the CEDAR Intake API.
- Add subcommands to `cmd/test_cedar_intake/main.go` - the `submit` subcommand runs the previous functionality of submitting to CEDAR from your local machine, while the `dump` subcommand creates JSON files locally as examples of the payloads that will be sent to CEDAR.
- ID field removed from lifecycle cost lines - from discussion with the CEDAR team, we decided to remove IDs from sub-entities; CEDAR will delete and regenerate sub-entities on their end when a top-level entity is updated.

## How to test this change

First, enable the `emit-to-cedar` feature flag for your EUA ID in the local environment in LaunchDarkly.

The easiest way to test this is to put a log/print statement in `pkg/cedar/intake/client.go`'s `publishIntakeObject()` method that logs when it's called. Run the full app locally, create an intake, request a draft business case (as an admin), and submit a draft business case (every field can be left empty); you should see the output from the log/print statement you added in the `easi` container logs.

For a more thorough test, connect to the VPN, make sure you have the `CEDAR_API_URL` and `CEDAR_API_KEY` environment variables set to valid values for CEDAR IMPL. Go through the same process of submitting a draft business case locally, then ask the CEDAR team in #oit-easi-cedar on Slack whether they received the payload.
